### DISCUSSION TITLE
Philosophy of Radix UI Themes

### DISCUSSION THREAD

**Discussion by Saadchr** [radix-ui/themes#232] [Q&A]
Hi,
I am trying to grasp the philosophy behind Radix UI Themes so I can fully understand it. I read somewhere that Radix Theme is not opinionated when it comes to styling.

So, except for the props exposed to the Radix UI primitives that are designed with the objective of harmony and should not be overridden by external styling, we are encouraged to come up with our own styling for all other styles, right?

So to fully use Radix Theme, I should first use the props, and if what I am looking for is not in the props, I could then come up with my own styling?
Thanks

**Comment by vladmoroz** [comment]
Hey, thanks for the question, you are thinking in the right direction.

"Should not be overridden by the external styling" might be a strong way to put it, but it's true that we don't optimise for that and instead provide the tools to build your own components in the form of the CSS variables and unstyled primitives that power everything in Radix Themes itself.

Also, by design, you can customise a lot by overriding the [CSS variables of the theme](https://github.com/radix-ui/themes/tree/main/packages/radix-ui-themes/src/styles/tokens), like fonts, colors, shadows, and other fundamental building blocks of the design.
